### PR TITLE
feat: component token rename

### PR DESCRIPTION
**Pull request by MadCcc**
- feat: component token naming rules
- dcos: update
- feat: modal token rename
- feat: rename result token
- feat: statsitic token rename
- feat: steps token remane

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
Follow https://github.com/ant-design/ant-design/wiki/API-Naming-rules#component-token

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    -       |
| 🇨🇳 Chinese |      -     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d38beca</samp>

This pull request adds a new feature called `component token` that allows users to customize the style of various components with a theme object. It updates the interface and usage of the token in the style files of the `button`, `drawer`, `message`, and `modal` components. It also adds demo files and documentation for each component to show how to use the feature.

### <samp>🤖 Generated by Copilot at d38beca</samp>

*  Add component token demos for button, drawer, message, and modal components to show how to customize their styles with theme objects ([link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-0ac9bc654a56e9c348611c0c703c15401ffc14693eb21f420e16dc8e35a805c4R1-R7), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-ef46974ad1140431488495c70259abb9806d06af6cb144ff5748354efa58a906R1-R16), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-80ce0ce5862cdbde65634fdeed1081c7812fabe85b736155e1be0c5da5760d16R50), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-447d93c86c05c00f9924d883b65877d29b1d52a00560746e69352a1d1c8af4ffR53), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-7a682405d0ce41c5d8a67bf6e2db533964ff5d1596ad3a89a40372e721bd70b5R1-R7), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-c0f84e80a996379f8d552a34d4255ec77e38de4f29e53bf7def873aea64db196R1-R17), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-7855c9de8eab66dde84ba7237f52b0fb88c917e3c8c78ada5fafa1e98c259721R37), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-bb2fbbde54ac7a64e8f607e35d8482d3150594c6466b025b29949ba82b675152R36), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-69e69019728952fee8314786b11ae390806088007615949747768aa81d0e4e74R1-R7), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-d3b10c09e0fea9643ada85519d76c70c53e1ede68daa19366ae3c57be794f6feR1-R20), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-de241cebff0141f80c65af1345bdfcebc23897cd4d0380161a09234079605dcbL31-R35), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-e806e8697391b559d536849650f46fe5e8368556ed033816ed2a8587675c9cb3R32), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8dca069cbd8e0bbb0f62b8dd7ca1be7c6072dd73aadd4701a74976b42865ad56R1-R7), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-9074e0b68d50f720bdba9aa718cf86d4aaf7ec13a484d12eb953e6e95bdb79faR1-R39), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-0868ee605478aef1b7a7e4508dce443964207f1dd966d82be9bf7e9b16a31aa5R39), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-7a204621cc9425007680c097ad01151df377979692bbc09eb2796025b85043f3R40))
* Simplify and standardize the naming of the token properties in the component token interfaces of button, drawer, message, and modal components to match the CSS properties ([link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L11-R11), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-cb2507ee40e1f2f8ba644202d9eb67d9eb5fc3d766b7ba3628703498a44225e3L7-R8), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL11-R19), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L11-R16), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486R33-R35))
* Modify the destructuring and use of the token properties in the style functions of button, drawer, message, and modal components to match the changes in the component token interfaces ([link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L22-R22), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L29-R29), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-cb2507ee40e1f2f8ba644202d9eb67d9eb5fc3d766b7ba3628703498a44225e3L34-R35), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-cb2507ee40e1f2f8ba644202d9eb67d9eb5fc3d766b7ba3628703498a44225e3L203-R203), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL27), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL40-R41), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL84-R86), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L158-R161), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L167-R167), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L188-R189), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L196-R200), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L220-R220), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L233-R233), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L300-R301))
* Modify the default values of the component tokens of button, drawer, message, and modal components to match the changes in the component token interfaces ([link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-e8f947f22fc66308c74c500faf1cfeab5e8eed33d2ab4a27718770da99224693L542-R542), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-cb2507ee40e1f2f8ba644202d9eb67d9eb5fc3d766b7ba3628703498a44225e3L225-R226), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL179-R190), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486R445-R447), [link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L456-R464))
* Add the import of the CSSProperties type from React to the `components/message/style/index.tsx` file to use it for the contentPadding token property ([link](https://github.com/ant-design/ant-design/pull/42184/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfR4))

### CODE REVIEW

File: docs/react/migrate-less-variables.zh-CN.md [hunk 1]
```diff
@@ -89,40 +92,40 @@ title: Less 变量迁移 Design Token
...
+| `@modal-header-title-line-height` | `titleLineHeight` | - |
+| `@modal-header-title-font-size` | `titleFontSize` | - |
 | `@modal-header-border-color-split` | - | 由于样式变化已废弃 |
 | `@modal-header-close-size` | - | 由于样式变化已废弃 |
-| `@modal-content-bg` | `modalContentBg` | - |
-| `@modal-heading-color` | `modalHeadingColor` | - |
-| `@modal-close-color` | `modalCloseIconColor` | - |
-| `@modal-footer-bg` | `modalFooterBg` | - |
+| `@modal-content-bg` | `contentBg` | - |
+| `@modal-heading-color` | `titleColor` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:104]: 这个titleColor 是只会影响 Title，不影响 closeIcon 的 color 是吧？所以才叫 titleColor？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 2]
```diff
@@ -89,40 +92,40 @@ title: Less 变量迁移 Design Token
...
+| `@modal-footer-bg` | `footerBg` | - |
 | `@modal-footer-border-color-split` | - | 由于样式变化已废弃 |
 | `@modal-footer-border-style` | - | 由于样式变化已废弃 |
 | `@modal-footer-padding-vertical` | - | 由于样式变化已废弃 |
 | `@modal-footer-padding-horizontal` | - | 由于样式变化已废弃 |
 | `@modal-footer-border-width` | - | 由于样式变化已废弃 |
 | `@modal-mask-bg` | `colorBgMask` | - |
 | `@modal-confirm-body-padding` | - | 由于样式变化已废弃 |
-| `@modal-confirm-title-font-size` | `modalHeaderTitleFontSize` | - |
+| `@modal-confirm-title-font-size` | `titleFontSize` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:114]: 这个是titleFontSize 会影响所有 title 吗？还是就只影响confirm的？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -89,40 +92,40 @@ title: Less 变量迁移 Design Token
...
 | Less 变量 | Component Token | 备注 |
 | --- | --- | --- |
 | `@notification-width` | `width` | - |
-| `@notification-padding` | `notificationPadding` | - |
-| `@notification-padding-vertical` | `notificationPaddingVertical` | - |
-| `@notification-padding-horizontal` | `notificationPaddingHorizontal` | - |
-| `@notification-margin-bottom` | `notificationMarginBottom` | - |
-| `@notification-margin-edge` | `notificationMarginEdge` | - |
-| `@notification-bg` | `notificationBg` | - |
+| `@notification-padding` | `paddingInlineStart` `paddingInlineEnd` `paddingBlockStart` `paddingBlockEnd` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:123]: 这里为啥要拆成 4 个？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 4]
```diff
@@ -89,40 +92,40 @@ title: Less 变量迁移 Design Token
...
+| `@notification-padding-vertical` | `paddingBlockStart` `paddingBlockEnd` | - |
+| `@notification-padding-horizontal` | `paddingInlineStart` `paddingInlineEnd` | - |
+| `@notification-margin-bottom` | `marginBottom` | - |
+| `@notification-margin-edge` | `marginInlineEnd` | - |
+| `@notification-bg` | `background` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:128]: @zombieJ 你说不会出现的场景在这里出现了，要命名为 bg 么？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 5]
```diff
@@ -154,43 +155,41 @@ title: Less 变量迁移 Design Token
...
 <!-- prettier-ignore -->
 | Less 变量 | Component Token | 备注 |
 | --- | --- | --- |
-| `@statistic-title-font-size` | `statisticTitleFontSize` | - |
-| `@statistic-content-font-size` | `statisticContentFontSize` | - |
-| `@statistic-font-family` | `statisticFontFamily` | - |
-
+| `@statistic-title-font-size` | `titleFontSize` | - |
+| `@statistic-content-font-size` | `contentFontSize` | - |
+| `@statistic-font-family` | `fontFamily` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:165]: 这个还需要么？直接就是 GlobalToken 里的了？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 6]
```diff
@@ -154,43 +155,41 @@ title: Less 变量迁移 Design Token
...
-| `@steps-small-icon-margin` | - | 由于样式变化已废弃 |
-| `@steps-dot-size` | `stepsDotSize` | - |
-| `@steps-dot-top` | - | 由于样式变化已废弃 |
-| `@steps-current-dot-size` | `stepsCurrentDotSize` | - |
-| `@steps-description-max-width` | `stepsNavContentMaxWidth` | - |
-| `@steps-nav-content-max-width` | - | 由于样式变化已废弃 |
-| `@steps-vertical-icon-width` | - | 由于样式变化已废弃 |
-| `@steps-vertical-tail-width` | - | 由于样式变化已废弃 |
-| `@steps-vertical-tail-width-sm` | - | 由于样式变化已废弃 |
+| `@process-tail-color` | `colorSplit` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:172]: 还有这个也是，如果是和 GlobalToken 的一样，是不是直接说使用GlobalToken 即可？不需要再标记为componentToken？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 7]
```diff
@@ -154,43 +155,41 @@ title: Less 变量迁移 Design Token
...
+| `@steps-nav-arrow-color` | `navArrowColor` | - |
+| `@steps-background` | `colorBgContainer` | - |
+| `@steps-icon-size` | `iconSize` | - |
```
Comment (arvinxx) [inline docs/react/migrate-less-variables.zh-CN.md:175]: 这个iconSize 和iconFontSize 有啥区别？

File: docs/react/migrate-less-variables.zh-CN.md [hunk 8]
```diff
@@ -89,40 +92,40 @@ title: Less 变量迁移 Design Token
...
+| `@modal-header-title-font-size` | `titleFontSize` | - |
 | `@modal-header-border-color-split` | - | 由于样式变化已废弃 |
 | `@modal-header-close-size` | - | 由于样式变化已废弃 |
-| `@modal-content-bg` | `modalContentBg` | - |
-| `@modal-heading-color` | `modalHeadingColor` | - |
-| `@modal-close-color` | `modalCloseIconColor` | - |
-| `@modal-footer-bg` | `modalFooterBg` | - |
+| `@modal-content-bg` | `contentBg` | - |
+| `@modal-heading-color` | `titleColor` | - |
+| `@modal-close-color` | `colorIcon` | - |
```
Comment (zombieJ) [inline docs/react/migrate-less-variables.zh-CN.md:105]: 这个 global token 和 component 语义规范有点不太一样。按照 component 规范似乎是叫 semantic: icon + css: color ?

Comment (zombieJ) [reply docs/react/migrate-less-variables.zh-CN.md:105]: 如果有多个 icon 来设置的话是不是这样子：
* closeIconColor
* resizeIconColor

File: docs/react/migrate-less-variables.zh-CN.md (hunk 2 above)
Comment (zombieJ) [reply docs/react/migrate-less-variables.zh-CN.md:114]: 两个名字一样的话看起来就是合起来了

File: docs/react/migrate-less-variables.zh-CN.md (hunk 4 above)
Comment (zombieJ) [reply docs/react/migrate-less-variables.zh-CN.md:128]: 规范里最小构成是 语义 + CSS ，这个看起来少了一个？

File: docs/react/migrate-less-variables.zh-CN.md (hunk 1 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:104]: 对

File: docs/react/migrate-less-variables.zh-CN.md (hunk 5 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:165]: 这个看了实现没有改过，直接用 global token 的就可以

File: docs/react/migrate-less-variables.zh-CN.md (hunk 6 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:172]: 是 global token， 但是迁移文档要说明一下

File: docs/react/migrate-less-variables.zh-CN.md (hunk 7 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:175]: iconSize 是容器大小，iconFontSize 是 icon 真实大小

File: docs/react/migrate-less-variables.zh-CN.md (hunk 4 above)
Comment (arvinxx) [reply docs/react/migrate-less-variables.zh-CN.md:128]: 问题就是这个 token 已经没有啥别的语义了

File: docs/react/migrate-less-variables.zh-CN.md (hunk 6 above)
Comment (arvinxx) [reply docs/react/migrate-less-variables.zh-CN.md:172]: 那如果是global token，是否需要单独一列说明是来自 GlobalToken ，不然看起来以为这个还是 componentToken

File: docs/react/migrate-less-variables.zh-CN.md (hunk 8 above)
Comment (arvinxx) [reply docs/react/migrate-less-variables.zh-CN.md:105]: > 这个 global token 和 component 语义规范有点不太一样。按照 component 规范似乎是叫 semantic: icon + css: color ?

这个是用了 GlobalToken。感觉文档里需要单独加一列 GlobalToken，告知迁移后的token 应该是用的ComponentToken，还是 GlobalToken

File: docs/react/migrate-less-variables.zh-CN.md (hunk 7 above)
Comment (arvinxx) [reply docs/react/migrate-less-variables.zh-CN.md:175]: 要不叫 iconContainerSize ？ 不然感觉没明白是啥意思

File: docs/react/migrate-less-variables.zh-CN.md (hunk 3 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:123]: 因为 closeBtn（absolute）会分别用到 top 和 right，所以这里全部拆开

File: docs/react/migrate-less-variables.zh-CN.md (hunk 4 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:128]: 会改成 componentBg

File: docs/react/migrate-less-variables.zh-CN.md (hunk 6 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:172]: 在备注里标明好了

File: docs/react/migrate-less-variables.zh-CN.md (hunk 7 above)
Comment (MadCcc) [reply docs/react/migrate-less-variables.zh-CN.md:175]: 这个要对应 Steps 的场景
image

File: docs/react/migrate-less-variables.zh-CN.md (hunk 3 above)
Comment (arvinxx) [reply docs/react/migrate-less-variables.zh-CN.md]: 这个 token 没体现出来是和这个closeIcon 的逻辑相关的唉？需要有这个关联关系吗

**Review by li-jia-nan — APPROVED** [review]

File: components/steps/style/custom-icon.ts [hunk 9]
```diff
@@ -12,11 +12,11 @@ const genStepsCustomIconStyle: GenerateStyle<StepsToken, CSSObject> = (token) =>
...
-          top: stepsIconCustomTop,
-          width: stepsIconCustomSize,
-          height: stepsIconCustomSize,
-          fontSize: stepsIconCustomFontSize,
-          lineHeight: `${stepsIconCustomSize}px`,
+          top: customIconTop,
+          width: customIconSize,
+          height: customIconSize,
+          fontSize: customIconFontSize,
+          lineHeight: `${customIconFontSize}px`,
```
Comment (Wxh16144) [inline components/steps/style/custom-icon.ts:19]: 重命名时改错了: https://github.com/ant-design/ant-design/issues/48641

**Pull request merged**